modu/source_repository/source_repository.py

```diff
@@ -86,17 +86,31 @@ def _parse_ast(self, of_file: FileInfo) -> tree_sitter.Tree:
             return ast
 
     def _discover_files(self) -> Dict[Path, FileInfo]:
+        # Get all files
+        files = [f for f in self.src_root.iterdir() if f.is_file()]
+        # Remove unwanted
+        ignored_files = self._get_ignores()
+        for ignore in ignored_files:
+            files = [n for n in files if not n.match(ignore)]
```

Comment:
So for each ignored file, we are iterating over the list of ignored files files and constructing a new list of files removing one file from the total list of files each iteration? if the goal is to remove all ignored files from the list of found files, I'm pretty sure the following works:

`[file for file in files if file not in ignored_files]`

Reply:
ignored files might be a bad name - will change it. ignored_files at the moment is a list of regex's that we use to match on. If they match a given file, we want to exclude them